[Bug] NeoForge 实例，离线登录「管理皮肤」选择的皮肤失效，且 authlib-injector 日志有异常反馈

### 问题描述

使用一个自定义了皮肤的离线账户 (用户名以 `Alex` 为例)，启动一个 NeoForge 实例，结果游戏中不展示用户设定的皮肤。

同时日志中有异常反馈。

注：日志已做脱敏处理，遮蔽了 SJMCL 自动生成的 UUID (以 `<uuid>` 展示)。

<details><summary>日志片段</summary>
<p>

```
[authlib-injector] [WARNING] An error occurred while processing request [/https/sessionserver.mojang.com/session/minecraft/profile/<uuid>]
java.io.UncheckedIOException: java.net.ConnectException: Connection refused: connect
	at moe.yushi.authlibinjector.yggdrasil.YggdrasilClient.queryProfile(YggdrasilClient.java:92)
	at moe.yushi.authlibinjector.httpd.QueryProfileFilter.handle(QueryProfileFilter.java:82)
	at moe.yushi.authlibinjector.httpd.URLProcessor$1.serve(URLProcessor.java:142)
	at moe.yushi.authlibinjector.internal.fi.iki.elonen.HTTPSession.execute(HTTPSession.java:246)
	at moe.yushi.authlibinjector.internal.fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:104)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.net.ConnectException: Connection refused: connect
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Net.java:546)
	at java.base/sun.nio.ch.Net.connect(Net.java:535)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:585)
	at java.base/java.net.Socket.connect(Socket.java:659)
	at java.base/java.net.Socket.connect(Socket.java:597)
	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:166)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:516)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:604)
	at java.base/sun.net.www.http.HttpClient.<init>(HttpClient.java:276)
	at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:380)
	at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:393)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1030)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:963)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:899)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:872)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1377)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1302)
	at moe.yushi.authlibinjector.util.IOUtils.http(IOUtils.java:49)
	at moe.yushi.authlibinjector.yggdrasil.YggdrasilClient.queryProfile(YggdrasilClient.java:90)
	... 5 more
```
```
[Download-1/WARN] [mojang/YggdrasilMinecraftSessionService]: Couldn't look up profile properties for <uuid>
com.mojang.authlib.exceptions.MinecraftClientException: Failed to read value Internal Server Error
	at com.mojang.authlib.minecraft.client.ObjectMapper.readValue(ObjectMapper.java:31) ~[authlib-9.0.75.jar:?] {}
	at com.mojang.authlib.minecraft.client.MinecraftClient.buildHttpException(MinecraftClient.java:169) ~[authlib-9.0.75.jar:?] {}
	at com.mojang.authlib.minecraft.client.MinecraftClient.readServiceResponse(MinecraftClient.java:147) ~[authlib-9.0.75.jar:?] {}
	at com.mojang.authlib.minecraft.client.MinecraftClient.getWithEtag(MinecraftClient.java:81) ~[authlib-9.0.75.jar:?] {}
	at com.mojang.authlib.minecraft.client.MinecraftClient.get(MinecraftClient.java:65) ~[authlib-9.0.75.jar:?] {}
	at com.mojang.authlib.yggdrasil.YggdrasilMinecraftSessionService.fetchProfileUncached(YggdrasilMinecraftSessionService.java:197) [authlib-9.0.75.jar:?] {}
	at com.mojang.authlib.yggdrasil.YggdrasilMinecraftSessionService.fetchProfile(YggdrasilMinecraftSessionService.java:167) [authlib-9.0.75.jar:?] {}
	at TRANSFORMER/minecraft@26.2/net.minecraft.client.Minecraft.lambda$new$0(Minecraft.java:405) [minecraft-client-patched-26.2.0.1-beta.jar:?] {}
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1789) [?:?] {}
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090) [?:?] {}
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614) [?:?] {}
	at java.base/java.lang.Thread.run(Thread.java:1474) [?:?] {}
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#unexpected-json-structure
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:521) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:1080) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:981) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:891) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:828) ~[gson-2.14.0.jar:?] {}
	at com.mojang.authlib.minecraft.client.ObjectMapper.readValue(ObjectMapper.java:29) ~[authlib-9.0.75.jar:?] {}
	... 11 more
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
See https://github.com/google/gson/blob/main/Troubleshooting.md#unexpected-json-structure
	at com.google.gson.stream.JsonReader.unexpectedTokenError(JsonReader.java:1831) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:507) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:510) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:1080) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:981) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:891) ~[gson-2.14.0.jar:?] {}
	at com.google.gson.Gson.fromJson(Gson.java:828) ~[gson-2.14.0.jar:?] {}
	at com.mojang.authlib.minecraft.client.ObjectMapper.readValue(ObjectMapper.java:29) ~[authlib-9.0.75.jar:?] {}
	... 11 more
```

</p>
</details> 

### 操作系统

Windows 11, version 25H2 (26200.8728)

### SJMCL 版本（可选）

[`b44b637`](https://github.com/UNIkeEN/SJMCL/commits/b44b6374b834ba564e951062991c81b9bbb7d6f6)

### 复现步骤

1. 设置一个离线账户，选一个<s>心仪的</s>皮肤。
2. 启动一个现代 NeoForge 实例 (我只用 1.21.1 和 26.2 进行了试验)
3. 建议打开日志窗口，方便同时观察日志

### 预期行为

原版和 Fabric 实例都可以加载皮肤，且没有异常日志。

我担心可能是 NeoForge 或者 authlib-injector 的问题，于是拿 HMCL 又跑了一通。然而 HMCL 一切正常 🤷‍♂️

### 环境信息（可选）

_No response_

### 补充信息（可选）

_No response_

--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -72,7 +72,7 @@ serde_json = "1.0"
 serde_with = "3.12.0"
 sjmcl-macros = { path = "./crates/sjmcl-macros" }
 sjmcl-types = { path = "./crates/sjmcl-types" }
-sha1 = "0.10.1"
+sha1 = { version = "0.10.1", features = ["oid"] }
 sha2 = "0.10.9"
 shlex = "1.3.0"
 smart-default = "0.7.1"

--- a/src-tauri/src/account/helpers/offline/yggdrasil_server.rs
+++ b/src-tauri/src/account/helpers/offline/yggdrasil_server.rs
@@ -56,7 +56,7 @@ fn get_public_key() -> String {
 }
 
 fn sign_data(data: &str) -> String {
-  let signing_key = SigningKey::<Sha1>::new_unprefixed(key_pair().0.clone());
+  let signing_key = SigningKey::<Sha1>::new(key_pair().0.clone());
   let signature = signing_key.sign(data.as_bytes());
   general_purpose::STANDARD.encode(signature.to_bytes())
 }
@@ -130,7 +130,7 @@ impl YggdrasilServer {
     let port = find_free_port(Some(18960)).unwrap(); // 饮水思源，爱国荣校
 
     Self {
-      root_url: format!("http://localhost:{}", port),
+      root_url: format!("http://127.0.0.1:{}", port),
       port,
       players: Arc::new(Mutex::new(vec![])),
     }

